pkg/csi/service/vanilla/controller.go

```diff
@@ -906,6 +921,464 @@ func (c *controller) getAccessibleTopologiesForDatastore(ctx context.Context, vc
+				}
+			}()
+
+			// Monitor CreateVolume task.
+			volumeMgr, err = common.GetVolumeManagerFromVCHost(ctx, c.managers, vcHost)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCode(log, codes.Internal, err.Error())
+			}
+			volumeInfo, faultType, err = volumeMgr.MonitorCreateVolumeTask(ctx,
+				&volumeOperationDetails, task, req.Name, c.managers.CnsConfig.Global.ClusterID)
+			if err != nil {
+				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to monitor task for volume %s on VC %q. Error: %+v", req.Name, vcHost, err)
+			}
+			volTaskAlreadyRegistered = true
+		}
+	}
+
+	var (
+		sharedDatastores    []*cnsvsphere.DatastoreInfo
+		topologyRequirement *csi.TopologyRequirement
+	)
+	// Get accessibility requirements.
+	topologyRequirement = req.GetAccessibilityRequirements()
+	if topologyRequirement == nil {
+		return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+			"accessibility requirements cannot be nil for a multi-VC environment")
+	}
+	// Club the accessibility requirements according to the VC they belong to.
+	vcTopologySegmentsMap, err := common.GetAccessibilityRequirementsByVC(ctx, topologyRequirement)
+	if err != nil {
+		return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+			"failed to club accessibility requirements by VC. Error: %+v", err)
+	}
+	log.Debugf("Topology accessibility requirements per VC are %+v", vcTopologySegmentsMap)
+
+	if !volTaskAlreadyRegistered {
+		// Check if topology domains have been provided in the vSphere CSI config secret.
+		// NOTE: We do not support kubernetes.io/hostname as a topology label.
+		if c.managers.CnsConfig.Labels.TopologyCategories == "" && c.managers.CnsConfig.Labels.Zone == "" &&
+			c.managers.CnsConfig.Labels.Region == "" {
+			return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+				"topology category names not specified in the vsphere config secret")
+		}
+
+		// Iterate through each VC and its accessibility requirements to try and create a volume.
+		// If it fails for any reason, move unto the next VC in list.
+		var topologySegmentsList []map[string]string
+		for vcHost, topologySegmentsList = range vcTopologySegmentsMap {
+			// Get VC instance.
+			vcenter, err = common.GetVCenterFromVCHost(ctx, c.managers.VcenterManager, vcHost)
+			if err != nil {
+				return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to get vCenter instance for host %q. Error: %+v", vcHost, err)
+			}
+
+			// If Storage policy is given, check if it exists in the VC. If not found, continue to next VC.
+			var storagePolicyID string
+			if scParams.StoragePolicyName != "" {
+				storagePolicyID, err = vcenter.GetStoragePolicyIDByName(ctx, scParams.StoragePolicyName)
+				if err != nil {
+					// TODO: Multi-VC: Catch NotFound error and continue only on NotFound error.
```

Comment:
Since we are returning the following error in govmomi in plain English and we do not have a locale for govmomi, we can rely on the error message.

https://github.com/vmware/govmomi/blob/9054158bb458f8c383422253709b56dc318ba2c0/pbm/client.go#L232

Reply:
Done